File: 백준/Silver/2503. 숫자 야구/숫자 야구.java
import java.io.*;
import java.util.*;


public class Main {
	
	static int n;
	static List<BaseBallData> inputData = new ArrayList<>();

	public static void main(String[] args) throws IOException {
		//입력
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		String s = br.readLine();
		n = Integer.parseInt(s);
		
		for (int i = 0; i<n; i++) {
			s = br.readLine();
			StringTokenizer st = new StringTokenizer(s, " ");
			int check = Integer.parseInt((st.nextToken()));
			int strike = Integer.parseInt((st.nextToken()));
			int ball = Integer.parseInt((st.nextToken()));
			
			BaseBallData baseBallData = new BaseBallData(check, strike, ball);
			inputData.add(baseBallData);
			
		}
		System.out.println(callBaseBall());
		
	}
	
	static int callBaseBall() {
		int result = 0;
		
		for (int i = 123; i<=987;i++) {
				if(!checkSameNum(i)) continue;
			
			
			int allTestPass = 0;
			
			for (int j = 0; j < n; j++) {
				int strikeCount = 0;
				int ballCount =0;
				
				BaseBallData current = inputData.get(j);
				String currentDataString = Integer.toString(current.check);
				String myDataString = Integer.toString(i);
				
				for (int k = 0; k < 3 ; k++) {
					if (currentDataString.charAt(k) == myDataString.charAt(k)) {
						strikeCount++;
					}
				}
				
				for (int h = 0; h < 3;h++) {
					for (int u = 0; u <3; u++) {
						if(myDataString.charAt(h) == currentDataString.charAt(u)) {
							if (h != u)
								ballCount++;
						}
					}
				}
				
				if (current.strike == strikeCount && current.ball == ballCount) {
					allTestPass++;
				}			
			}
			if (allTestPass ==n) {
				result ++;
			}
		}
		return result;
	}
	
	
	static boolean checkSameNum(int num) {
		
		String numString = Integer.toString(num);
		
		if (numString.charAt(0) == numString.charAt(1)) {
			return false;
		}
		if (numString.charAt(1) == numString.charAt(2)) {
			return false;
		}
		if (numString.charAt(2) == numString.charAt(0)) {
			return false;
		}
		if (numString.charAt(0) == '0' || numString.charAt(1) == '0' || numString.charAt(2) == '0')
            return false;
		
		return true;
		
	}
	
	static class BaseBallData{
		int check;
		int strike;
		int ball;
		
		public BaseBallData(int check, int strike, int ball) {
			this.check = check;
			this.strike = strike;
			this.ball = ball;
		}
	}

}

File: 백준/Silver/2503. 숫자 야구/숫자 야구.py
n = int(input())


numbers = [list(map(str,input().split())) for _ in range (n)]
    
answer = 0

for a in range(1,10):
    for b in range(1,10):
        for c in range(1,10):
            #각 케이스 마다 볼카운트, 스트라이크 카운트 확인
            tmp = 0
            
            # 서로 다른 세 자리수
            if (a == b or b == c or c == a):
                # print(str(a)+str(b)+str(c))
                continue

            # 조건 대입하기
            for array in numbers:
                check = list(array[0])
                strike = int(array[1])
                ball = int(array[2])
                ball_count = 0
                strike_count = 0

                #스트라이크 계산기
                if (a == int(check[0])):
                    strike_count += 1
                if (b == int(check[1])):
                    strike_count += 1
                if (c == int(check[2])):
                    strike_count += 1
                if (strike != strike_count):
                    break

                #볼 계산기
                if (a == int(check[1]) or a == int(check[2])):
                    ball_count +=  1
                if (b == int(check[0]) or b == int(check[2])):
                    ball_count +=  1
                if (c == int(check[0]) or c == int(check[1])):
                    ball_count +=  1
                if (ball != ball_count):
                    break

                #일치하면 더해주기
                tmp += 1
            if tmp == n:
                answer += 1
                
print(answer)

